### PR TITLE
Quicker error reporting when using `sandbox.sh`

### PR DESCRIPTION
**Pull request by jumaffre**
After issues such as #3757, we now report an error early (5secs vs. 60secs previously) when a node does not start up successfully when using `sandbox.sh`. 

For example, setting the host log level to `trace` for a release build:
```
$ sandbox.sh -l trace --js-app-bundle ~/git/CCF/tests/js-authentication/ 
Setting up Python environment...
Python environment successfully setup
[15:59:30.004] Virtual mode enabled
[15:59:30.005] Starting 1 CCF node...
[15:59:35.126] Error! Some nodes ran into issues:
[15:59:35.126] - Node [0]:
terminating with uncaught exception of type std::logic_error: No enclave file found at ./libjs_generic.virtual.so

[15:59:35.126] Please raise a bug if the issue is unexpected: https://github.com/microsoft/CCF/issues/new?assignees=&labels=bug&template=bug_report.md&title=Unexpected%20error%20when%20running%20sandbox%20script
```

### CODE REVIEW

File: tests/infra/network.py [hunk 1]
```diff
@@ -72,8 +72,10 @@ class StartupSnapshotIsOld(Exception):
     pass
 
 
-class NodeShutdownError(Exception):
-    pass
+class NetworkShutdownError(Exception):
+    def __init__(self, msg, errors=None):
+        self.errors = errors
+        super().__init__(msg)
```
Comment (achamayou) [inline tests/infra/network.py:78]: Seems odd to initialize the superclass last rather than first?

**Comment by achamayou** [comment]
@jumaffre this is a good idea, but we could raise much quicker than the newly introduced timeout, as soon as the subprocess returns. Waiting on the PEM is reasonable if the process is still going, but not if it's already terminated.

File: tests/infra/remote.py [hunk 2]
```diff
@@ -884,8 +889,8 @@ def suspend(self):
     def resume(self):
         self.remote.resume()
 
-    def get_startup_files(self, dst_path, timeout=FILE_TIMEOUT):
-        self.remote.get(self.pem, dst_path, timeout=timeout)
+    def get_startup_files(self, dst_path, timeout=FILE_TIMEOUT_S):
+        self.remote.get(self.pem, dst_path, timeout=0.1)
```
Comment (jumaffre) [inline tests/infra/remote.py:893]:
```suggestion
        self.remote.get(self.pem, dst_path, timeout=REMOTE_STARTUP_TIMEOUT_S)
```

**Comment by jumaffre** [comment]
> @jumaffre this is a good idea, but we could raise much quicker than the newly introduced timeout, as soon as the subprocess returns. Waiting on the PEM is reasonable if the process is still going, but not if it's already terminated.

I thought about this too and I don't think we want to introduce an additional stall when we start the node process to check for successful startup as this would introduce a significant delay to all node launches, even successful ones. Hence adding the check around an existing operation, with a much shorter timeout and retry logic so that it doesn't impact the run time when the node startups successfully, while reporting startup failures quickly.

**Comment by achamayou** [comment]
@jumaffre I don't understand, I think we can check for the return code in the same loop that we check for the PEM? That shouldn't add any delay?

**Review by achamayou — APPROVED** [review]

**Pull request merged**